### PR TITLE
docs: add tutorials of fine-tune on a custom dataset

### PR DESCRIPTION
**Pull request by XixinYang**
Thank you for your contribution to the MindCV repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: docs/en/how_to_guides/finetune_with_a_custom_dataset.md [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+            └── ....
+```
+
+./example/finetune.py integrates the whole training pipeline, from pre-processing to the establishment and training of the model: `create_dataset` -> `create_transforms` -> `create_loader` -> `create_model` ->..., thus, the dataset with the adequate file directory structure can be sent directly to the fine-tuning script to start the subsequent processes including loading dataset and model training by running `python ./example/finetune.py --data_dir=./aircraft/data/images/` command. For custom datasets, please note that<font color=DarkRed> the dataset parameter in the configuration file must be set to an empty string `""` </font> in advance.
+
+#### Read Dataset online
+
+Offline data reading takes up extra local disk space to store the newly generated data files. Therefore, when the local storage space is insufficient or the data cannot be backed up to the local environment, the local data files cannot be read using `create_dataset` directly, you can write a function to read the dataset in an online way.
+
+Here's how we generate an iterable object that stores the images of the training set:
```
Comment (SamitHuang) [inline docs/en/how_to_guides/finetune_with_a_custom_dataset.md:125]: I think the source dataset defined below is a mappable dataset instead of an iterable dataset( which should have __iter__ method.

**Review by SamitHuang — COMMENTED** [review]
Very comprehensive:)

File: docs/en/how_to_guides/finetune_with_a_custom_dataset.md [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+	- Next, the iterable object is constructed using the `__getitem__` function.
+- After writing the ReadDataset class, we can pass it the path of the annotation file to instantiate it, and load it as a dataset that can be read by the model through [`mindspore.dataset.GeneratorDataset`](https://www.mindspore.cn/docs/zh-CN/r2.0/api_python/dataset/mindspore.dataset.GeneratorD ataset.html#mindspore.dataset.GeneratorDataset). Note that <font color=DarkRed>the parameter `column_names` must be set to be ["image", "label"]</font> for subsequent reading by other functions. What we've got now is supposed to be the same as what's generated by `create_dataset`.
+
+
+```python
+import numpy as np
+from mindspore.dataset import GeneratorDataset
+
+
+class ReadDataset:
```
Comment (SamitHuang) [inline docs/en/how_to_guides/finetune_with_a_custom_dataset.md:141]: ReadDataset -> ImageClsDataset?

File: docs/en/how_to_guides/finetune_with_a_custom_dataset.md [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+
+```bash
+python finetune.py --config=./configs/mobilenetv3/mobilnet_v3_small_ascend.yaml --data_dir=./aircraft/data --pretrained=True
+```
+
+When fine-tuning mobilenet v3-small based on Aircraft dataset, this tutorial mainly made the following changes to the hyper-parameters:
+
+| Hyper-parameter | Before     | After                |
```
Comment (SamitHuang) [inline docs/en/how_to_guides/finetune_with_a_custom_dataset.md:203]: Before -> Pretrain
After -> Fine-tune

File: docs/en/how_to_guides/finetune_with_a_custom_dataset.md [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+
+# read names of parameters in FC layer
+classifier_names = [_model_pretrained_cfgs[args.model]["classifier"] + ".weight",
+                    _model_pretrained_cfgs[args.model]["classifier"] + ".bias"]
+
+# prevent parameters in network(except the classifier) from updating
+for param in network.trainable_params():
+    if param.name not in classifier_names:
+        param.requires_grad = False
+```
```
Comment (SamitHuang) [inline docs/en/how_to_guides/finetune_with_a_custom_dataset.md:240]: It should be better to wrap these code of freezing backbone into a function, allowing easy integration for other datasets/tasks.

```python
def freeze_backbone(net, cfg):
   ...
   return net

```

File: docs/en/how_to_guides/finetune_with_a_custom_dataset.md [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+
+# read names of network layers
+freeze_layer=["features."+str(i) for i in range(7)]
+
+# prevent parameters in the first 7 layers of the network from updating
+for param in network.trainable_params():
+    for layer in freeze_layer:
+    	if layer in param.name:
+        	param.requires_grad = False
+```
```
Comment (SamitHuang) [inline docs/en/how_to_guides/finetune_with_a_custom_dataset.md:257]: Aren't these params already frozen by the code in #### Linear Probe ?

File: docs/en/how_to_guides/finetune_with_a_custom_dataset.md [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+                    "lr": lr_scheduler}，
+                  {"order_params": network.trainable_params()}]
+
+optimizer = create_optimizer(params_lr_group,
+                             opt=args.opt,
+                             lr=lr_scheduler,
+                             ...)
+```
+
+#### Set Learning Rate for Specific Layers
```
Comment (SamitHuang) [inline docs/en/how_to_guides/finetune_with_a_custom_dataset.md:288]: Duplicate with line 259. 

We may combine these sections for param grouping into one, e.g.
```
### Param Grouping: Set Different Learning Rates for Feature Layers and Classifier Layers 
```

File: docs/en/how_to_guides/finetune_with_a_custom_dataset.md (hunk 5 above)
Comment (XixinYang) [reply docs/en/how_to_guides/finetune_with_a_custom_dataset.md:257]: This is the example code of `Freeze Part of the Feature Network`, which is another circumstance different from `Linear Probe`.

File: docs/en/how_to_guides/finetune_with_a_custom_dataset.md (hunk 1 above)
Comment (XixinYang) [reply docs/en/how_to_guides/finetune_with_a_custom_dataset.md:125]: How about "an random-accessible dataset object that stores the images of the training set and realize a map from indices to data samples"? the "random-accessible dataset" term is from mindspore site.

File: docs/en/how_to_guides/finetune_with_a_custom_dataset.md [hunk 7]
```diff
@@ -0,0 +1,357 @@
...
+        for image in image_label[label]:
+            image_name = image + ".jpg"
+            shutil.copy(images_path + image_name, label_folder + image_name)
+
+
+images_path = "./aircraft/data/images/"
+subset_name = "trainval"
+annotation_file_path = "./aircraft/data/images_variant_trainval.txt"
+extract_images(images_path, subset_name, annotation_file_path)
+```
```
[the remark this answers is not in the document]
Comment (XixinYang) [reply docs/en/how_to_guides/finetune_with_a_custom_dataset.md:93]: added that.

File: docs/en/how_to_guides/finetune_with_a_custom_dataset.md [hunk 8]
```diff
@@ -0,0 +1,357 @@
...
+    └── ....
+```
+
+The folder "images" contains all the 10,000 images, and the airplane types and subset names of each image are recorded in images_variant_*.txt. When this dataset is used for fine-tuning, the training set is usually set by annotation file: images_variant_trainval.txt. Hence, the training set should contain 6667 images and the test set should contain 3333 images after the dataset has been split.
+
+## Data Preprocessing
+
+### Read Custom Dataset
+
+For custom datasets, you can either organize the dataset file directory locally into a tree structure similar to ImageNet, and then use the function `create_dataset` to read the dataset (offline way), or directly read all the images into an iterable object, replacing the file splitting and the `create_dataset` steps (online way).
```
[the remark this answers is not in the document]
Comment (geniuspatrick) [reply docs/en/how_to_guides/finetune_with_a_custom_dataset.md:27]: Since the structure of downstream datasets can be diverse, to be able to handle this diversity, the following two approaches are common practice:

Offline processing: Manually reorganize data into a standard format for loading by existing interfaces. The keyword **_offline_** is reflected in the process of manually formatting the dataset.

Online processing: Building new interfaces to be able to load datasets in specific formats. The keyword **_online_** is reflected in the fact that the newly constructed interface is parsing the data structure during runtime.

File: docs/en/how_to_guides/finetune_with_a_custom_dataset.md (hunk 6 above)
Comment (geniuspatrick) [reply docs/en/how_to_guides/finetune_with_a_custom_dataset.md:288]: Just make the subtitile more specific. `for classifier(267)` or `for specific layers(278)` are two excellent examples. Replace `for specific layers(278)` with `for whatever layers you want`? Still, it is an exciting example.

**Review by geniuspatrick — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]
